02-technical-guides/aggregate-provider/main.ipynb

```diff
@@ -0,0 +1,1564 @@
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th>VaR95</th>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>BUSINESS_UNIT</th>\n",
+       "      <th>SUB_BUSINESS_UNIT</th>\n",
+       "      <th>TRADING_DESK</th>\n",
+       "      <th>ccy</th>\n",
+       "      <th></th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>Equity</th>\n",
+       "      <th>Foreign Equity</th>\n",
+       "      <th>Hermes</th>\n",
+       "      <th>EUR</th>\n",
+       "      <td>-3,647,071.70</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>Forex</th>\n",
+       "      <th>Developed Market</th>\n",
+       "      <th>Hades</th>\n",
+       "      <th>EUR</th>\n",
+       "      <td>-3,554,955.65</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>Rates &amp; Credits</th>\n",
+       "      <th>Domestic Rates &amp; Credit</th>\n",
+       "      <th>Hermes</th>\n",
+       "      <th>EUR</th>\n",
+       "      <td>-4,681,608.40</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "                                                                  VaR95\n",
+       "BUSINESS_UNIT   SUB_BUSINESS_UNIT       TRADING_DESK ccy               \n",
+       "Equity          Foreign Equity          Hermes       EUR  -3,647,071.70\n",
+       "Forex           Developed Market        Hades        EUR  -3,554,955.65\n",
+       "Rates & Credits Domestic Rates & Credit Hermes       EUR  -4,681,608.40"
+      ]
+     },
+     "execution_count": 16,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "cube.query(\n",
+    "    m[\"VaR95\"],\n",
+    "    levels=[l[\"BUSINESS_UNIT\"], l[\"SUB_BUSINESS_UNIT\"], l[\"TRADING_DESK\"], l[\"ccy\"]],\n",
+    "    filter=(l[\"AsOfDate\"] == \"2021-03-01\"),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "be8263f3-fe7f-4f3c-9fad-c72a0b84dc05",
+   "metadata": {},
+   "source": [
+    "### Explaining query plans"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c4a2c263-ac7a-4f93-ae0c-f6d087d55933",
+   "metadata": {},
+   "source": [
+    "We can understand how the query is executed with [`cube.explain_query`](https://docs.atoti.io/latest/lib/atoti/atoti.cube.html#atoti.Cube.explain_query).  \n",
+    "\n",
+    "We replace the `query` function with `explain_query` and re-run the query with levels defined in the aggregate provider.  We will see that _PartialPrimitiveAggregateRetrieval_ is used in the query plan. "
```

Comment:
PartialPrimitiveAggregatesRetrieval

Also, it might be helpful to add something like "Under Retrievals" before "(w)e will see that [. . .]" just because it is halfway down the query explanation.